src/blob/core.rs

```diff
@@ -198,6 +203,7 @@ where
             self.file.clone(),
             bincode::serialized_size(&self.header)?,
             K::LEN as usize,
+            self.validate_data_during_index_regen,
```

Comment:
I suggest to pass this as a parameter to `raw_records`. If someone will start using this function just to read record headers, it would be surprise, that it also validates data. So it is better to give explicit control over the behavior though parameter